### PR TITLE
Allow Kokkos with deprecated code disabled

### PR DESCRIPTION
**Pull request by masterleinad**
This pull request allows building `DTK` if the underlying `Kokkos` library has deprecated code disabled.

`Intrepid2` still needs to be fixed upstream, though.

### CODE REVIEW

**Comment by masterleinad** [comment]
See https://github.com/trilinos/Trilinos/pull/6539.

File: packages/Meshfree/test/tstNearestNeighborOperator.cpp [hunk 1]
```diff
@@ -264,7 +264,8 @@ TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( NearestNeighborOperator, mixed_clouds,
         double ref_value = round( target_points( i, 0 ) / Lx * nx ) * Lx / nx;
         if ( ref_value == Lx * comm_size )
             ref_value -= Lx / nx;
-        TEST_FLOATING_EQUALITY( target_values_host( i, 0 ), ref_value, 1e-14 );
+        TEST_FLOATING_EQUALITY( target_values_host.access( i, 0 ), ref_value,
+                                1e-14 );
```
Comment (dalg24) [inline packages/Meshfree/test/tstNearestNeighborOperator.cpp]: Just drop the 2nd argument.

File: packages/Meshfree/src/DTK_DetailsNearestNeighborOperatorImpl.hpp [hunk 2]
```diff
@@ -85,15 +85,14 @@ struct NearestNeighborOperatorImpl
...
         Kokkos::parallel_for(
             DTK_MARK_REGION( "get_source_values" ),
             Kokkos::RangePolicy<ExecutionSpace>( 0, n_imports ),
             KOKKOS_LAMBDA( int i ) {
-                for ( int j = 0; j < (int)source_values.dimension_1(); ++j )
-                    buffer_values( i, j ) =
-                        source_values( import_source_indices( i ), j );
+                for ( int j = 0; j < (int)source_values.extent( 1 ); ++j )
+                    buffer_values.access( i, j ) =
+                        source_values.access( import_source_indices( i ), j );
```
Comment (dalg24) [inline packages/Meshfree/src/DTK_DetailsNearestNeighborOperatorImpl.hpp]: Comment that using `View::access()` is just a workaround and that we could/should write two specializations of the kernel.

File: packages/Meshfree/src/DTK_DetailsNearestNeighborOperatorImpl.hpp [hunk 3]
```diff
@@ -129,9 +128,9 @@ struct NearestNeighborOperatorImpl
             DTK_MARK_REGION( "set_target_values" ),
             Kokkos::RangePolicy<ExecutionSpace>( 0, n_imports ),
             KOKKOS_LAMBDA( int i ) {
-                for ( int j = 0; j < (int)target_values.dimension_1(); ++j )
-                    target_values( import_target_indices( i ), j ) =
-                        import_source_values( i, j );
+                for ( int j = 0; j < (int)target_values.extent( 1 ); ++j )
+                    target_values.access( import_target_indices( i ), j ) =
+                        import_source_values.access( i, j );
```
Comment (dalg24) [inline packages/Meshfree/src/DTK_DetailsNearestNeighborOperatorImpl.hpp]: Comment here as well.

**Pull request merged**